.github/workflows/publish.yml

```diff
@@ -0,0 +1,27 @@
+name: Publish
```

Comment:
I got confused by this name and the file name. It's called "publish", but in fact it does both checking and publishing

Reply:
yeah, can probably find better names, maybe: `Build and publish` for now?

Reply:
sure!